### PR TITLE
added proxy complience

### PR DESCRIPTION
**Pull request by udarrr**
1. Added proxy for http, https requests for cases when the tool behind corporate proxy and system env HTTP,HTTPS were specified
2. Dropped support node12 as http and https are node14 tools

### CODE REVIEW

**Comment by udarrr** [comment]
@christian-bromann do you mind review it and if ok then trigger release ?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node-version: [12, 14, 16, 18, 20]
```
Comment (christian-bromann) [inline .github/workflows/test.yml:14]: If we drop support for Node.js v12 we have to do a major release

File: HISTORY.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# 9.6.0 (2023-10-21)
```
Comment (christian-bromann) [inline HISTORY.md:1]: if we do a major release this should be `10.0.0`
```suggestion
# 10.0.0 (2024-06-24)
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
One question on the Node.js v12 drop

File: .github/workflows/test.yml (hunk 1 above)
Comment (udarrr) [reply .github/workflows/test.yml:14]: yep it makes sense

**Comment by udarrr** [comment]
> One question on the Node.js v12 drop

done

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**